### DISCUSSION TITLE
How to correctly define custom type for tstzrange?

### DISCUSSION THREAD

**Discussion by JesperWe** [jackc/pgx#1858] [General]
I have understood that the `tstzrange` pgtype equivalent is `pgtype.Range[time.Time]`, and that if I declare my own type for this pgtype will not be able to recognize it. ("It currently cannot automatically detect the underlying type of renamed structs")

I'm trying to do exactly this: Implement my own renamed type

```
type TimeRange pgtype.Range[time.Time]
```

Then I am trying to implement custom TextValuer/Scanner

```
func (t TimeRange) TextValue() (pgtype.Text, error) {
        ....stuff....
	return pgtype.Text{
		String: "...."
	}, nil
}

func (t TimeRange) ScanText(v pgtype.Text) error {
        ....stuff....
	return nil
}
```

Here TextValue() is called and seems to work, but ScanText is never called and I get errors like 

> can't scan into dest[5]: cannot scan tstzrange (OID 3910) in binary format into *b.TimeRange

What am I missing?

**Comment by jackc** [comment]
You need to implement `pgtype.RangeValuer` and `pgtype.RangeScanner` instead of `pgtype.TextValuer` and `pgtype.TextScanner`.

**Comment by JesperWe** [comment]
Ok, will try, thanx!
